File: 2022/01/a.cpp
#include <iostream>
#include <queue>
#include <sstream>

using namespace std;

void solution_one() {
  int ans = 0;
  int cur = 0;
  string line;
  while(getline(cin, line)) {
		if(line.empty()) {
			ans = max(ans, cur);
			cur = 0;
		} else {
			istringstream iss(line);
			int n;
			iss >> n;
			cur += n;
		}
	}
	ans = max(ans, cur);
	cout << ans << endl;
}

void solution_two() {
	priority_queue<int> pq;
	int cur(0);
	string line;
	while(getline(cin, line)) {
		if(line.empty()) {
			pq.push(cur);
			cur = 0;
		} else {
			istringstream iss(line);
			int n;
			iss >> n;
			cur += n;
		}
	}
	pq.push(cur);
	int ans(0);
	for(int i=0; i<3; i++) {
		ans += pq.top();
		pq.pop();
	}
	cout << ans << endl;
}

int main() {
	solution_two();
}

File: 2022/01/a.py
f = open("input.txt", "r")


def solution_one():
    max = 0
    cur = 0
    for line in f:
        if line == '\n':
            max = max if max > cur else cur
            cur = 0
        else:
            cur += int(line)

    max = max if max > cur else cur
    print(max)


def solution_two():
    ans_arr = []
    cur = 0
    for line in f:
        if line == '\n':
            ans_arr.append(cur)
            cur = 0
        else:
            cur += int(line)
    ans_arr.append(cur)
    ans_arr.sort()
    print(sum(ans_arr[-3:]))


solution_two()
